### PR TITLE
Make the 'ble' and 'wifi' options mutually exclusive

### LINKED ISSUE #19: Code which is generated does not compile
Hi! 

Running following command: `esp-generate --chip esp32c3 -o ble -o wifi -o embassy -o alloc -o probe-rs -o wokwi -o ci xiao-no-std-esp-gen` leads to a project which does not compile. 

Compiler output:
```
WARNING: use --release
  We *strongly* recommend using release profile when building esp-hal.
  The dev profile can potentially be one or more orders of magnitude
  slower than release, and may cause issues with timing-senstive
  peripherals and/or devices.

   Compiling xiao-no-std-esp-gen v0.1.0 (D:\Software\esp32_rust\xiao-no-std-esp-gen)
error[E0277]: the trait bound `EspWifiInitFor: EspWifiTimerSource` is not satisfied
   --> src\bin\async_main.rs:31:9
    |
29  |     let _init = esp_wifi::init(
    |                 -------------- required by a bound introduced by this call
30  |         esp_wifi::EspWifiInitFor::Wifi,
31  |         esp_wifi::EspWifiInitFor::Ble,
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `esp_wifi::IntoAnyTimer` is not implemented for `EspWifiInitFor`, which is required by `EspWifiInitFor: EspWifiTimerSource`
    |
    = help: the following other types implement trait `esp_wifi::IntoAnyTimer`:
              Alarm<'_, T, DM, COMP, UNIT>
              AnyTimer
              esp_hal::timer::timg::Timer<T, DM>
    = note: required for `EspWifiInitFor` to implement `EspWifiTimerSource`
note: required by a bound in `esp_wifi::init`
   --> C:\Users\ckren\.cargo\registry\src\index.crates.io-6f17d22bba15001f\esp-wifi-0.10.1\src/lib.rs:406:17
    |
404 | pub fn init(
    |        ---- required by a bound in this function
405 |     init_for: EspWifiInitFor,
406 |     timer: impl EspWifiTimerSource,
    |                 ^^^^^^^^^^^^^^^^^^ required by this bound in `init`

error[E0061]: this function takes 4 arguments but 5 arguments were supplied
   --> src\bin\async_main.rs:29:17
    |
29  |     let _init = esp_wifi::init(
    |                 ^^^^^^^^^^^^^^
...
32  |         timg1.timer0,
    |         ------------ unexpected argument #3 of type `esp_hal::timer::timg::Timer<TimerX<TIMG1>, Blocking>`
    |
note: function defined here
   --> C:\Users\ckren\.cargo\registry\src\index.crates.io-6f17d22bba15001f\esp-wifi-0.10.1\src/lib.rs:404:8
    |
404 | pub fn init(
    |        ^^^^
help: remove the extra argument
    |
31  -         esp_wifi::EspWifiInitFor::Ble,
32  -         timg1.timer0,
31  +         esp_wifi::EspWifiInitFor::Ble,
    |
```

I also mentioned this in the matrix chat:
https://matrix.to/#/!LdaNPfUfvefOLewEIM:matrix.org/$Cqev3HK8Nc5K5MYE3pwcDjRd_ijKB3PmmWr3Ueyqk6Q?via=matrix.org&via=tchncs.de&via=envs.net

### PR DESCRIPTION
**Pull request by jessebraham**
Closes #19

### CODE REVIEW

**Review by SergioGasquez — APPROVED** [review]
LGTM! Thanks!

File: src/main.rs [hunk 1]
```diff
@@ -391,17 +391,24 @@ fn process_options(args: &Args) {
...
-                    "Error: Option {:?} is not supported for chip {:?}",
+                    "Error: Option '{}' is not supported for chip {}",
                     option, args.chip
                 );
+                process::exit(-1);
             }
         }
     }
+
+    if args.option.contains(&String::from("ble")) && args.option.contains(&String::from("wifi")) {
```
Comment (bjoernQ) [inline src/main.rs:408]: That's fine but I was hoping we could detect such a situation more generally by checking the `disables` field in `GeneratorOption`

But given this is not a general-purpose library to create project generators it's probably unnecessary to deal with that

**Review by bjoernQ — APPROVED** [review]
Thanks!

Feel free to explore a way to make detection of mutual exclusive features more general or not (if not I might someday ... )

File: src/main.rs (hunk 1 above)
Comment (jessebraham) [reply src/main.rs:408]: I did consider a more thorough solution, but for the time being this at least solve the linked issue. Can revisit it at a later date.

**Pull request merged**